protoutil/conversion.go

```diff
@@ -22,7 +22,8 @@ import (
 	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/structpb"
-	"gopkg.in/yaml.v2"
+
+	ghodssyaml "github.com/ghodss/yaml"
```

Comment:
Is there a reason we're switching packages? `gopkg.in/yaml` is the most popular one. I'm not really supportive of bringing in a bespoke YAML library.

Reply:
Seems like its sitting on top of gopkg.in/yaml and is good if you need to support both yaml and json, see https://web.archive.org/web/20150812020634/http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang/

Reply:
gopkgin.yaml doesn't work here, it can't handle maps.  gopkgin.yaml is also used by k8s sig https://pkg.go.dev/sigs.k8s.io/yaml#section-readme

Reply:
Yes, `gopkg.in/yaml` unmarshals yaml nests into `map[interface{}]interface{}` rather than `map[string]interface{}`, which in turn causes problem with json marshaling.

Reply:
For the record, I've used `github.com/ghodss/yaml` before and know it's being used in some other well-known OSS projects. I think it's easier to use than `gopkg.in/yaml`

Reply:
I'm not sure I understand. It works fine with the gopkg.in/yaml.v3: https://gist.github.com/sethvargo/b11a0e22886edcc3c61e95a5a375f588

Reply:
I was about to ask - have we tried v3 yet?

Reply:
done